### ISSUE TITLE
feedback - feature requests

### ISSUE THREAD

**Issue by Fischmuetze** [netdcy/FlowVision#6]
Hi,
I like this app - it's very impressive fast! Congrats!

Things that I miss / would nice to have - suggestions only of course 

- recursive view for folder / subfolders
- stepless or more fine granulate zoom for thumbnails by a slider in the tool or status bar
- a reduced line spacing for the folders in the sidebar 
- a reduced or configurable font size in the sidebar
- a reduced line spacing for the thumbnails
- a reduced or configurable font size for the thumbnails view

right hand of the thumbnails an info bar 
- for file infos, and meta data
- for a preview

strange  behaviour:
- App quits by pressing ESC (I would expecting only in preview mode, that it returned back to the browser mode)

Thank you!

**Comment by netdcy** [comment]
Thank you for your feedback and suggestions. I'm very glad that you like this application. 

Most of the suggestions you mentioned are about details, which I have considered and chosen during the design process. Therefore, please provide sufficient reasons for the changes. Allowing custom settings is an option, but implementing this is somewhat cumbersome. So unfortunately, most points cannot be changed immediately, but you can compile it yourself; compiling this is not difficult.

Regarding Homebrew, it seems to only accept applications with an Apple signature, so it cannot be added at this time. I'm not sure what you mean by the recursive view of the folder? As for the continuous change of thumbnail size, the reason it cannot be implemented is that the waterfall layout needs to exactly fill the width of the view, so it must be discrete values.

**Comment by Fischmuetze** [comment]
> a reduced line spacing for the folders in the sidebar
> a reduced or configurable font size in the sidebar

both are using not the same line spacing and font size as macOS self - see compare the tree view of Apple Mail and FlowView. It could be that the larger line spacing of FlowView is due to the use of a larger font. It certainly looks unusual in my eyes. See the screenshot in the background Mail. Would be great if this would be using the same line spacing as the most in macOS

![fontsize](https://github.com/user-attachments/assets/ef709056-e63f-463b-817b-8249b54750b8)

Another small UI point is the thickness of the dividing line between the folder area and the view area. Normally in macOS this divider is just a thin line, but in FlowView it is this kind of wide bar 

> Regarding Homebrew, it seems to only accept applications with an Apple signature, so it cannot be added at this time. 

No, that's not the case ... Apps like yours are also available there. I have installed some of them like your via Homebrew

> I'm not sure what you mean by the recursive view of the folder? 

If you select a folder to view its contents, not only the im items in this folder are displayed, but also all images of all subfolders of this selected folder. Other viewers like XnView MP or Phoenix Slides have this option (but both are not really Mac like)


Anyway a big thank you for your effort

**Comment by netdcy** [comment]
Thank you for your careful observation.

It seems you are not using the default system sidebar font size(https://www.youtube.com/watch?v=dee3IYBQ9CQ). The system's medium size is 13, and the small size is 11. In this program, the sidebar uses 13, and the thumbnail view uses 12. Additionally, it seems the Finder's sidebar font color is not pure black; I will confirm this.

> Another small UI point is the thickness of the dividing line between the folder area and the view area. Normally in macOS this divider is just a thin line, but in FlowView it is this kind of wide bar

Regarding the separator bar, using SwiftUI makes it easy to achieve an appearance that is completly identical to Finder (I tried, but SwiftUI cannot implement a bug-free waterfall layout, so I had to rewrite it using AppKit). Using AppKit, it seems only the current effect can be achieved. If the separator bar is set hidden, there will be no shadow at the boundary, which might look even worse.

> If you select a folder to view its contents, not only the im items in this folder are displayed, but also all images of all subfolders of this selected folder. Other viewers like XnView MP or Phoenix Slides have this option (but both are not really Mac like)

I understand. Implementing this is actually simple, so this feature might be added in the future.

**Comment by dreamer-cloud** [comment]
Loving this app, throwing in few suggestions, some are interlinked with OP's:

- Ability to hide filenames in waterfall view: this one is not that critical in light theme, but with dark theme on file names on the thumbnail card become somewhat prominent enough to case tiny bit of distraction.

- Ability to sort order of folders in sidebar - also feels pretty crucial when dealing with large amount of folders, in order to have ability to dive into latest updated folder first. For now workaround is adding lately updated folders manually to Favorites list.

- Ability to prevent ESC key from hiding the whole window - not sure whether i'm the only one falling for it frequently, but feel it's pretty counter-intuitive for ones used to macos patterns

**Comment by Fischmuetze** [comment]
> Ability to hide filenames in waterfall view: this one is not that critical in light theme, but with dark theme on file names on the thumbnail card become somewhat prominent enough to cause tiny bit of distraction.

yes but then a tooltip with the filename should be available for hovering or an info palette for the selected item

> Ability to sort order of folders in sidebar - also feels pretty crucial when dealing with large amount of folders, in order to have ability to dive into latest updated folder first. For now workaround is adding lately updated folders manually to Favorites list.

no that's strange - it's the tree of the file system - this would struggle into issues understanding the structure and for drag and drop actions.
But of course would be nice to have the bookmarks in the side bar  and/or a list of the 10 most recent folders 

> Ability to prevent ESC key from hiding the whole window - not sure whether i'm the only one falling for it frequently, but feel it's pretty counter-intuitive for ones used to macos patterns

yes that's a very strange behavior, as this not only hides the app, but completely terminates it

**Comment by dreamer-cloud** [comment]
> yes but then a tooltip with the filename should be available for hovering or an info palette for the selected item

or maybe it would be enough just to see filename in tab bar/title bar upon clicking on it/opening image full view like it is now

> no that's strange - it's the tree of the file system - this would struggle into issues understanding the structure and for drag and drop actions.

woudn't it be logical to follow patterns of any Open/Save/Finder window where you able to sort the tree they way you want? don't really understand why one is able to sort images in a folder in any descending/ascending way but not the file tree

**Comment by Fischmuetze** [comment]
@netdcy 

> It seems you are not using the default system sidebar font size(https://www.youtube.com/watch?v=dee3IYBQ9CQ). The system's medium size is 13, and the small size is 11. In this program, the sidebar uses 13, and the thumbnail view uses 12. Additionally, it seems the Finder's sidebar font color is not pure black; I will confirm this.

It would be great if the scaling of the font size and the vertical spacing of the icons in the sidebar would follow the scaling behavior of the macOS. Means if the user choose the small size in the macOS Preferences it would follow this setting

**Comment by netdcy** [comment]
The behavior of the Esc key has been changed in the re-uploaded compiled version.

I also think that having the directory tree follow the sorting order would cause confusion, as people usually just need to sort the images in the current folder by date. The modification date of the folder itself becomes meaningless on macOS due to system files like .DS_Store.

As for hiding file names and adjusting the directory tree text size according to system settings, it is somewhat complex. Maybe it will be done in the future when there is time.

**Comment by marjue12255** [comment]
Hello netdcy
Very nice app. Good work. There are not much viewers out there which are capable of HDR and color management. 
Is there a way to hide to top status bar in fullscreen mode. so that only the image is visible? 

Greetings
Marcus

**Comment by helmut72** [comment]
> recursive view for folder / subfolders

This! A common timeline view like in smartphones photo apps would be cool, but should scan all subfolders from the current folder, since my photos are also sorted in (sub)folders.

Thank you.

**Comment by elendirx** [comment]
This app is already installable by homebrew

**Comment by greatvovan** [comment]
A request to @Fischmuetze to update the content of the original post as it's now out of date:

- ~Installable by Homebrew~ – implemented
- ~recursive view for folder / subfolders~  – implemented
- ~stepless or more fine granulate zoom for thumbnails by a slider in the tool or status bar~ – won't fix
- a reduced line spacing for the folders in the sidebar
- a reduced or configurable font size in the sidebar
- a reduced line spacing for the thumbnails
- a reduced or configurable font size for the thumbnails view

right hand of the thumbnails an info bar

- for file infos, and meta data
- for a preview

strange behaviour:
- ~App quits by pressing ESC (I would expecting only in preview mode, that it returned back to the browser mode)~ – fixed.
